lib/puppet/provider/package/yum.rb

```diff
@@ -156,6 +156,11 @@ def install
     operation = :install
 
     case should
+    when :latest
+      # use update_command when using latest for updating packages
+      operation = update_command
```

Comment:
Also, isn't the code at line 175/180 supposed to handle that? Why isn't it working? Oh, because latest doesn't compare correctly.

This should at least check that `self.query` returns a value. This seems like the logic should be moved down to the else block, and be checked as part of version comparison.